### ISSUE TITLE
Rule Request: prefer assign operators

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#902]
Prefer assign operators (`+=`, `-=`, `*=`, `/=`) over doing the operation and assigning:

```swift
var foo = 0

// should trigger
foo = foo + 2

// shouldn't trigger
foo += 2
```

A concrete example: https://github.com/realm/SwiftLint/pull/893#discussion_r89646409
